Add documentation for pagination

conwaycolin opened the pull request.

bjcoleman commented:
Let's make the flask server serve up the HTML file with `render_template`.   

I don't remember, is having the files in `static` and `templates` enough for `Flask` to serve up both files?  If not, make whatever changes are necessary.

conwaycolin committed:
documentation for pagination

conwaycolin committed:
Cursor-based pagination

conwaycolin committed:
Add query modifications

conwaycolin committed:
Implement pagination for API

Added functionality for the API to process
the page query parameter necessary for pagination.

Pagination is not functional it is just handled
appropriatley by the API.

conwaycolin committed:
Merge branch 'main' of https://github.com/cs334s24/Mirrulations_Search into pagination

conwaycolin committed:
Implement pagination for search_dockets

Pagination works but tests don't currently work

Commit:
Make tests compliant with pagination

Added necessary parameters to tests to be compliant
with pagination.

conwaycolin committed:
Update documentation
